include/circt/Dialect/HW/HWOpInterfaces.td

```diff
@@ -0,0 +1,72 @@
+//===- HWOpInterfaces.td - Operation Interfaces ------------*- tablegen -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This describes the HW operation interfaces.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef CIRCT_DIALECT_HW_HWOPINTERFACES
+#define CIRCT_DIALECT_HW_HWOPINTERFACES
+
+include "mlir/IR/OpBase.td"
+
+def HWModuleLike : OpInterface<"HWModuleLike"> {
+  let cppNamespace = "circt::hw";
+  let description = "Provide common  module information.";
+
+  let methods = [
+    InterfaceMethod<"Check whether the module is publicly visible",
+    "bool", "isPublic", (ins),
+    /*methodBody=*/[{
+      return ::mlir::SymbolTable::getSymbolVisibility($_op) ==
```

Comment:
Any particular reason this is a methodBody while all the other ones are defaultImplementation?

Reply:
Based on my understanding, `defaultImplementation` can be overriden? Since this consults the symbol table, I don't want `FModuleOp`s to provide a different implementation.